HotelBase/HotelExampleBookRoomCantFail.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using HotelBooking;
+
+namespace HotelBase
+{
+    public class HotelExampleBookRoomCantFail : HotelBaseClass, IHotel
+    {
+        public HotelExampleBookRoomCantFail(IBookingService bookingService, IPaymentService paymentService, ILogger logger) : base(bookingService, paymentService, logger)
+        {
+
+        }
+
+        internal override bool CheckPrice(double price, DateTime date)
+        {
+            if (double.IsNaN(price) || date == default(DateTime))
+                return false;
+            return true;
+        }
+
+        internal override bool SendEmail(string email)
+        {
+            if (email == null)
+                return false;
+            return true;
+        }
+
+        public ReservationResult Reserve(DateTime date, double price, int creditCardNumber, string email)
+        {
+            bool priceValid = CheckPrice(price, date);
+            bool paymentMade = MakePayment(creditCardNumber, price);
+            bool emailSent = SendEmail(email);
+            bool roomBooked = BookRoom(date);
+            string reservationNumber;
+            bool success;
+            if (roomBooked)
```

Comment:
Should consider:

In this task you have to add a new hotel with some small changes.  In other words the booking "algorithm" looks the very similar to other classes, but in this case you have to write it and its test from scratch. It will be nice if you could consider changing a code in a way, which allow me to add next hotel  without writing booking algorithm one more time. It will be cool to  only configure steps order and its failing policy/option (I was not sure how to name this process).

Reply:
I've added generic algorithm to the base class